jdaviz/app.py

```diff
@@ -887,6 +827,197 @@ def _get_all_subregions(mask, spec_axis_data):
 
         return regions
 
+    def get_subsets(self, subset_name=None, spectral_only=False,
+                    spatial_only=False, object_only=False):
+        """
+        Returns all branches of glue subset tree in the form that subset plugin can recognize.
+
+        Parameters
+        ----------
+        subset_name : str
+            The subset name.
+        spectral_only : bool
+            Return only spectral subsets.
+        spatial_only : bool
+            Return only spatial subsets.
+        object_only : bool
+            Return only object relevant information and
+            leave out the region class name and glue_state.
+
+        Returns
+        -------
+        data : dict
+            A dict with keys representing the subset name and values as astropy regions
+            objects
+        """
+
+        dc = self.data_collection
+        subsets = dc.subset_groups
+        # TODO: Use global display units
+        # units = dc[0].data.coords.spectral_axis.unit
+        viewer = self.get_viewer("spectrum-viewer")
```

Comment:
This line is giving me error in Imviz when I have Subsets and I call `imviz.app.get_subsets()`. There is no doc updates to go with this, so I am not sure how I am supposed to test this in Imviz. Please advise. thanks!

```
File jdaviz/app.py:858, in Application.get_subsets(self, subset_name, spectral_only, spatial_only, object_only)
    855 subsets = dc.subset_groups
    856 # TODO: Use global display units
    857 # units = dc[0].data.coords.spectral_axis.unit
--> 858 viewer = self.get_viewer("spectrum-viewer")
    859 data = viewer.data()
    860 # Set axes labels for the spectrum viewer

File jdaviz/app.py:589, in Application.get_viewer(self, viewer_reference)
    564 def get_viewer(self, viewer_reference):
    565     """
    566     Return a `~glue_jupyter.bqplot.common.BqplotBaseView` viewer instance.
    567     This is *not* an ``IPyWidget``. This is stored here because
   (...)
    587         The viewer class instance.
    588     """
--> 589     return self._viewer_by_reference(viewer_reference)

File jdaviz/app.py:1409, in Application._viewer_by_reference(self, reference)
   1394 """
   1395 Viewer instance by reference defined in the yaml configuration file.
   1396 
   (...)
   1405     The viewer class instance.
   1406 """
   1407 viewer_item = self._viewer_item_by_reference(reference)
-> 1409 return self._viewer_store[viewer_item['id']]

TypeError: 'NoneType' object is not subscriptable
```

Reply:
p.s. Didn't @bmorris3 have things like this to prevent hardcoding "spectrum-viewer"?

https://github.com/spacetelescope/jdaviz/blob/5510486a3a47adb9b955de26747d3662b77fbf9d/jdaviz/configs/cubeviz/helper.py#L20